File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numToIndex = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in numToIndex:
                return [i, numToIndex[complement]]
            numToIndex[nums[i]] = i
            

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> indices;
        for(int i = 0; i < nums.size(); i++){
            int complement = target - nums[i];
            if(indices.find(complement) == indices.end()){
                indices[nums[i]] = i; 
            }
            else{
                return {indices[complement], i};
            }
        }
        return {};
    }
};